hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyLocationInfoGroup.java

```diff
@@ -89,17 +90,27 @@ public long getVersion() {
     return version;
   }
 
+  /**
+   * Use this expensive method only when absolutely needed!
+   * It creates a new list so it is not an O(1) operation.
+   * Use getLocationLists() instead.
+   * @return a list of OmKeyLocationInfo
+   */
   public List<OmKeyLocationInfo> getLocationList() {
     return locationVersionMap.values().stream().flatMap(List::stream)
         .collect(Collectors.toList());
   }
 
+  public Collection<List<OmKeyLocationInfo>> getLocationLists() {
+    return locationVersionMap.values();
+  }
+
   public long getLocationListCount() {
     return locationVersionMap.values().stream().mapToLong(List::size).sum();
   }
 
-  public List<OmKeyLocationInfo> getLocationList(Long versionToFetch) {
-    return new ArrayList<>(locationVersionMap.get(versionToFetch));
+  public Collection<OmKeyLocationInfo> getLocationList(Long versionToFetch) {
```

Comment:
this is a minor issue but allocating an ArrayList is an O(n) operation. I don't see this one coming up in flamegraph (which means it's probably minimal) but it's a good practice to reduce it to O(1)

Reply:
According to call hierarchy this is only used in write path, once per key (from `BlockOutputStreamEntryPool#addPreallocateBlocks`).  Based on that, I think we could omit this change now.

Return type does not need to be changed anyway.

Reply:
ok omit this for now.